src/utils/resources/vmi/utils/cloud-init.ts

```diff
@@ -1,23 +1,38 @@
+import { load } from 'js-yaml';
+
 import { V1VirtualMachineInstance } from '@kubevirt-ui/kubevirt-api/kubevirt';
 
+type CloudinitUserDataObject = {
+  user?: string;
+  password?: string;
+  passwd?: { users: { name: string }[] };
+};
+
 export const getCloudInitCredentials = (
   vmi: V1VirtualMachineInstance,
-): { user: string; password: string } => {
+): { user?: string; password?: string; users?: string[] } => {
   const cloudInitVolume = vmi?.spec?.volumes?.find(
     (volume) => volume?.cloudInitNoCloud || volume?.cloudInitConfigDrive,
   );
 
-  const cloudInit = cloudInitVolume?.cloudInitNoCloud || cloudInitVolume?.cloudInitConfigDrive;
+  // KubeVirt supports the cloud-init NoCloud and ConfigDrive data sources
```

Comment:
im not sure this comment is needed :)

Reply:
It is not needed but I liked to add it as it makes clearer what the following code is about ;)

Reply:
when looking on line 15, i think this self explanatory :) volume?.cloudInitNoCloud || volume?.cloudInitConfigDrive

Reply:
Well, the implementation that followed that line was not so self explanatory